code/game/objects/items/plushes.dm

```diff
@@ -367,13 +417,22 @@
 	if(mood_message)
 		desc += mood_message
 
+GLOBAL_LIST_INIT(valid_plushie_paths, valid_plushie_paths())
+/proc/valid_plushie_paths()
+	. = list()
+	for(var/i in subtypesof(/obj/item/toy/plush))
+		var/obj/item/toy/plush/abstract = i
+		if(initial(abstract.abstract_type) == i)
+			continue
+		. += i
+
 /obj/item/toy/plush/random
 	name = "Illegal plushie"
 	desc = "Something fucked up"
 	var/blacklisted_plushes = list(/obj/item/toy/plush/carpplushie/dehy_carp, /obj/item/toy/plush/awakenedplushie, /obj/item/toy/plush/random)
 
 /obj/item/toy/plush/random/Initialize()
-	var/newtype = pick(subtypesof(/obj/item/toy/plush) - typecacheof(blacklisted_plushes))
+	var/newtype = prob(CONFIG_GET(number/snowflake_plushie_prob))? /obj/item/toy/plush/random_snowflake : pick(GLOB.valid_plushie_paths)
```

Comment:
Congratulations, you brought back dehydratated carps to the random plushies pool.
Yes, I'm a bit sarcastic but I don't quite get why it'd need such a fancy hard to maintain global list.